src/main/java/vazkii/botania/client/integration/jei/orechid/OrechidRecipeWrapper.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * This class was created by <codewarrior0>. It's distributed as
+ * part of the Botania Mod. Get the Source Code in github:
+ * https://github.com/Vazkii/Botania
+ * <p/>
+ * Botania is Open Source and distributed under the
+ * Botania License: http://botaniamod.net/license.php
+ */
+package vazkii.botania.client.integration.jei.orechid;
+
+import mezz.jei.api.ingredients.IIngredients;
+import mezz.jei.api.recipe.IRecipeWrapper;
+import net.minecraft.init.Blocks;
+import net.minecraft.item.ItemBlock;
+import net.minecraft.item.ItemStack;
+import net.minecraftforge.oredict.OreDictionary;
+import vazkii.botania.api.BotaniaAPI;
+import vazkii.botania.client.integration.jei.JEIBotaniaPlugin;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class OrechidRecipeWrapper implements IRecipeWrapper, Comparable<OrechidRecipeWrapper> {
+    public String oreDictKey;
+    public int weight;
+
+    static ItemStack inputStack;
+    List<List<ItemStack>> outputStacks;
+
+    public ItemStack getInputStack() {
+        if (inputStack == null) inputStack = new ItemStack(Blocks.STONE, 64);
+        return inputStack;
+    }
+
+    public List<List<ItemStack>> getOutputStacks() {
+        if (outputStacks == null) {

```

Comment:
why the weird lazy-init stuff (same for the input and in the other class)? I'd rather just compute it all up front and have it be final

Reply:
I wasn't sure if everything below that would be available at static init time. I've had problems before with doing things in static initializers that should have been done later.

Reply:
But I should have just put it all in the normal constructor. Right.